### PR TITLE
Fix bugs in OpenACC directives and revise machine files

### PR DESCRIPTION
**Pull request by hyungyukang**
Fix bugs in OpenACC directives:
- Change an openacc directive in mpas_ocn_diagnostics.F to avoid
getting stuck on pm-gpu.

Fixes E3SM-Project/E3SM#6127

[BFB]

### CODE REVIEW

**Comment by ndkeen** [comment]
I'm seeing changes here that will have impact for other cases. We may need to find a way to make changes that will only affect MPAS

**Comment by hyungyukang** [comment]
> I'm seeing changes here that will have impact for other cases. We may need to find a way to make changes that will only affect MPAS. If MPAS needs a flag, then maybe should only be applied to those sources. If you need `-acc`, maybe only use that flag for the sources required. And please don't change the GPU bindings unless we all agree upon those changes as they impact all cases.

@ndkeen , Thanks for your instructions. I'm going to revert some changes I made. It seems like the best way to add `-acc` to `nvidiagpu` in MPAS is to copy and modify `Depends.pgigpu.cmake`. What do you think about this?

File: cime_config/machines/Depends.pm-gpu.nvidiagpu.cmake [hunk 1]
```diff
@@ -2,9 +2,13 @@ list(APPEND REDUCE_OPT_LIST
   homme/src/share/derivative_mod_base.F90
 )
 
+# add accelerator/gpu flags for MPAS files
```
Comment (grnydawn) [inline cime_config/machines/Depends.pm-gpu.nvidiagpu.cmake:5]: According to @ndkeen's comment, this setting might be placed in the Depends file, not in this cmake file, in order to limit the scope of impact to specified MPAS files.

File: cime_config/machines/cmake_macros/nvidiagpu.cmake [hunk 2]
```diff
@@ -4,14 +4,15 @@ set(MPIFC "ftn")
...
+
 string(APPEND CPPDEFS " -DFORTRANUNDERSCORE -DNO_R16 -DCPRNVIDIA")
 string(APPEND CPPDEFS_DEBUG " -DYAKL_DEBUG")
 if (compile_threaded)
   string(APPEND CMAKE_C_FLAGS " -mp")
 endif()
 string(APPEND CMAKE_C_FLAGS_RELEASE " -O2")
 string(APPEND CMAKE_C_FLAGS_DEBUG " -g")
-string(APPEND CMAKE_Fortran_FLAGS " -i4 -Mstack_arrays -Mextend -byteswapio -Mflushz -Kieee -DHAVE_IEEE_ARITHMETIC -Mallocatable=03 -DNO_R16 -traceback")
+string(APPEND CMAKE_Fortran_FLAGS " -i4 -Mstack_arrays -Mextend -byteswapio -Mflushz -Kieee -DHAVE_IEEE_ARITHMETIC -Mallocatable=03 -DNO_R16 -traceback -acc")
```
Comment (grnydawn) [inline cime_config/machines/cmake_macros/nvidiagpu.cmake:15]: According to @ndkeen's comment, this setting might be placed in the Depends file, not in this cmake file, in order to limit the scope of impact to specified MPAS files.

**Review by grnydawn — CHANGES_REQUESTED** [review]
I also think that we may need to restrict the scope of impact from this changes within MPAS. Please see my comments in my review. I prefer to have a GPU bind suggested in this PR instead of "none". But it may need to coordinate with other developers who may impact to the GPU bind change.

**Comment by amametjanov** [comment]
Rebased the branch onto latest master and checked that this PR still fixes the hang on PM-GPU: https://my.cdash.org/viewTest.php?buildid=2460796 .

**Comment by hyungyukang** [comment]
@amametjanov , thanks for rebasing this PR!

@grnydawn , many changes I made in this PR were duplicates of #6103 (merged) except for `mpas_ocn_diagnostics.F`, so they disappeared by rebasing this PR.

@ndkeen , this PR fixes #6116.

**Review by ndkeen — APPROVED** [review]
Now need https://github.com/E3SM-Project/E3SM/pull/6166 to allow building, but after that, this fix allows the test to run

**Review by grnydawn — APPROVED** [review]
@amametjanov, the changes that Hyun made in "mpas_ocn_diagnostics.F." look good to me.

However, my test case (compset: GMPAS-IAF, res: T62_EC30to60E2r2) on Hyun's code branch (hkang/ocean/pm-gpu-fix) fails during building SPIO with the following error message:

```
"/global/u2/y/youngsun/repos/github/E3SM/externals/scorpio/src/clib/pioc.c", line 1282: error: too many arguments in invocation of macro "adios2_init"
          ios->adiosH = adios2_init(ios->adios_comm, adios2_debug_mode_on);
```

Since this error is not related to the changes in the MPAS code, I believe this PR is ready to be merged.

**Pull request merged**